probe-rs/src/rtt.rs

```diff
@@ -115,27 +115,45 @@ impl Rtt {
 
     // Minimum size of the ControlBlock struct in target memory in bytes with empty arrays
     const MIN_SIZE: usize = Self::O_CHANNEL_ARRAYS;
+    const MIN_SIZE_64: usize = Self::O_CHANNEL_ARRAYS_64;
 
     // Offsets of fields in target memory in bytes
     const O_ID: usize = 0;
     const O_MAX_UP_CHANNELS: usize = 16;
     const O_MAX_DOWN_CHANNELS: usize = 20;
+    const O_MAX_DOWN_CHANNELS_64: usize = 24;
     const O_CHANNEL_ARRAYS: usize = 24;
+    const O_CHANNEL_ARRAYS_64: usize = 32;
 
     fn from(
         core: &mut Core,
         memory_map: &[MemoryRegion],
         // Pointer from which to scan
-        ptr: u32,
+        ptr: u64,
         // Memory contents read in advance, starting from ptr
         mem_in: Option<&[u8]>,
     ) -> Result<Option<Rtt>, Error> {
+        let is_64bit = match core.instruction_set()? {
```

Comment:
This would be a lot nices as `if matches!(core.instruction_set()?, probe_rs_target::InstructionSet::A64)`. We only need to list the ones that are 64-bit. I realize the downside is that adding new architectures are immediately treated as 32-bit, though, so maybe instead this logic could be moved into a function like `InstructionSet::width()`?

Reply:
> so maybe instead this logic could be moved into a function like InstructionSet::width()?

In this code, I want to know the size of the target integer(32bit or 64bit).
I'm checking it from the InstructioSet now, but I think there has to be a better than the current method.
Could you please tell me if you know of any good ideas?

Reply:
I found that Armv8a state has an `is_64_bit` flag, so I changed this code to use it.
https://github.com/probe-rs/probe-rs/pull/2296/commits/008aa3a6aa532ede92c027e52944b51d20c11c26